compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -585,7 +585,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
         if !errors.is_empty() {
             self.adjust_fulfillment_errors_for_expr_obligation(&mut errors);
-            self.err_ctxt().report_fulfillment_errors(&errors, self.inh.body_id);
+            self.err_ctxt().report_fulfillment_errors(
+                &errors,
+                self.inh.body_id.map(|body_id| body_id.hir_id.owner.def_id.to_def_id()),
```

Comment:
`self.body_id.to_def_id()` is more accurate, since we have a body def id in `FnCtxt`